src/main/java/com/aws/greengrass/shadowmanager/model/configuration/ShadowSyncConfiguration.java

```diff
@@ -225,37 +226,37 @@ private static void processThingShadowSyncConfiguration(Set<ThingShadowSyncConfi
         Validator.validateThingName(thingName);
 
         ThingShadowSyncConfiguration syncConfiguration;
-        boolean syncClassicTopic = true;
-        for (Map.Entry<String, Object> configObjectEntry : thingConfig.entrySet()) {
-            switch (configObjectEntry.getKey()) {
-                case CONFIGURATION_CLASSIC_SHADOW_TOPIC:
-                    syncClassicTopic = Coerce.toBoolean(configObjectEntry.getValue());
-                    break;
-                case CONFIGURATION_NAMED_SHADOWS_TOPIC:
-                    if (configObjectEntry.getValue() instanceof List) {
-                        List<String> namedShadows = Coerce.toStringList(configObjectEntry.getValue());
-                        for (String namedShadow : namedShadows) {
-                            Validator.validateShadowName(namedShadow);
-                            syncConfiguration = ThingShadowSyncConfiguration.builder()
-                                    .thingName(thingName)
-                                    .shadowName(namedShadow)
-                                    .build();
-                            syncConfigurationSet.add(syncConfiguration);
-                        }
-                    } else {
-                        throw new InvalidConfigurationException(String.format(UNEXPECTED_TYPE_FORMAT,
-                                configObjectEntry.getKey(), configObjectEntry.getClass().getTypeName()));
-                    }
-                    break;
-                default:
-                    // Do nothing here since we want to be lenient with unknown fields.
-                    break;
+        boolean addedOnInteraction = Coerce.toBoolean(
+                thingConfig.getOrDefault(CONFIGURATION_ADDED_ON_INTERACTION_TOPIC, false)
+        );
+
+        Object namedShadowsObj = thingConfig.getOrDefault(CONFIGURATION_NAMED_SHADOWS_TOPIC, null);
+        if (namedShadowsObj != null) {
+            if (!(namedShadowsObj instanceof List)) {
+                throw new InvalidConfigurationException(
+                        String.format(UNEXPECTED_TYPE_FORMAT, CONFIGURATION_NAMED_SHADOWS_TOPIC,
+                                namedShadowsObj.getClass().getTypeName()));
+            }
+            List<String> namedShadows = Coerce.toStringList(namedShadowsObj);
+            for (String namedShadow : namedShadows) {
+                Validator.validateShadowName(namedShadow);
+                syncConfiguration = ThingShadowSyncConfiguration.builder()
+                        .thingName(thingName)
+                        .shadowName(namedShadow)
+                        .addedOnInteraction(addedOnInteraction)
```

Comment:
added is past tense, you might want to call it `addOnInteraction` since the event has not take place.

Reply:
I left it as `addedOnInteraction` for now but I can change it to `addOnInteraction`. I'm not sure I understand your point fully - my thought process was that the event already took place since the thing shadow is in the component configuration.